text/0757-default-modifier-manager.md

```diff
@@ -0,0 +1,276 @@
+>
+> <cite style="display: block; text-align: right;">@ember/render-modifier's README</site>
+
+
+An example:
+
+```js
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+export default class Example extends Component {
+  @tracked numberOfClicks = 0
+
+  // when the number of clicks is odd, add a JS movement effect / animation
+  wiggly = (element) => {
+    if (this.numberOfClicks % 2 === 0) {
+      return;
+    }
+
+    // event callback
+    let animate = () => {
+      element.animate({ /* ... */ });
+    }
+
+    element.addEventListener('mousemove', animate);
+
+    // Cleanup
+    return () => element.removeEventListener('mousemove', animate);
+  }
+
+  increment = () => this.numberOfClicks++;
+}
+```
+```hbs
+Num clicks: {{this.numberOfClicks}}
+
+<button
+  type="button"
+  {{this.wiggly}}
+  {{on 'click' this.increment}}
+>
+  Try to click me
+</button>
+```
+
+This demonstrates that a modifier can:
+ - be defined as needed, there is no need for the modifier to live in its own file or a separate folder
+ - semantically encapsulate setup and teardown logic -- the returned function is the teardown logic
+ - have access to the element without leaking DOM details until the component class
+ - react to changes to tracked data consumed within the setup stage of the modifier (`this.numberOfClicks`)
+
+[gh-ember-modifier]: https://github.com/ember-modifier/ember-modifier/
+[gh-render-modifiers]: https://github.com/emberjs/ember-render-modifiers
+[gh-ecgutt]: https://github.com/tracked-tools/ember-could-get-used-to-this
+[rfc-779]:https://github.com/emberjs/rfcs/pull/779
+
+This pairs nicely with the [First-Class Component Templates RFC](rfc-779);
+where modifiers may be defined in the same file as a template-only component.
+
+<details><summary>Example</summary>
+
+<!--
+  If you're reading this RFC non-rendered, ignore the "jsx" language tag.
+  "jsx" is incorrect, and isn't perfect, but "gjs" syntax highlighting
+  hasn't been added to GitHub's highlighting yet.
+-->
+
+This could be an entire component
+
+```jsx
+const draggable = element => {
+  element.addEventListener( /*... */ );
+
+  return () => element.removeEventListener( /* ... */ );
+};
+
+<template>
+  <div {{draggable}}>text content A</div>
+  <div {{draggable}}>text content B</div>
+  <div {{draggable}}>text content C</div>
+  <div {{draggable}}>text content D</div>
+</template>
+```
+_Note: `<template>` is experimental syntax, and should not be be the focus of this example_
+
+
+</details>
+
+
+## Detailed design
+
+_A Default Manager is not something that can be chosen by the user, but is baked in to the framework
+as a default so that a user doesn't have to build something to use a non-framework-specific variant
+of the three constructs: Helpers, Modifiers, and Components._
```

Comment:
Couple quick notes on this:

- You're not advocating for a default component manager anymore, so "three constructs" and "Components" probably need to go.
- The actual motivation is that the user doesn't have to use special framework-provided tools to build these, but can just use normal JS constructs in places where it makes sense.

Reply:
Updated!, thanks!